### DISCUSSION TITLE
Using Relational DBs

### DISCUSSION THREAD

**Discussion by ricocsharp** [OrchardCMS/OrchardCore#12169] [General]
I tried to create Data tables and "one to many" relations between them using the examples given in Dojo Course 2, then I realized that it is kind of deprecated so, which is the way to stablish this relations and to create customa data tables if it is steel possible off course?

**Comment by ns8482e** [comment]
I don't know what's Dojo course 2 - I guess may be you are talking about orchard 1.x.

In orchard core - content is a document stored in single document record.  However there are some ways to achieve 1-many relations 

- Single document : You can create content item with BagPart. Here  parent/child items stored as single document  ( e.g. order and order items) this is useful when you have fewer child items.
- Multiple documents  -  you can create content item with ListPart - here parent/child items stored as independent documents e.g. Blog, BlogPost
- Multiple documents - via reference - you can create multiple content items and you can use ContentPickerField to link content items.

**Reply by ricocsharp** [reply]
Thanks a lot U have been of great help!! Now Do U know were to see programmatic examples of Multiple documents with this parent/child feature?

**Reply by ns8482e** [reply]
checkout the Blog theme - uses ListPart

**Reply by ricocsharp** [reply]
I'm using the headless CMS recipe , so int the front I'm using angular SPA that will cosume via graphql my data I have some clasic CMS content management and for this Contant parts and documents are ok, but I will usa some modules to manage business data that if it could be possible I need it to use relations between data

**Reply by ns8482e** [reply]
For relational data that’s not cms document - if you prefer you can define graph ql or you can define rest apis

**Comment by hishamco** [comment]
@ns8482e Dojo is a series of Orchard / Orchard Core courses by Lombiq

@ricocsharp please checkout Dojo 3 which is target Orchard Core, https://orcharddojo.net/orchard-training/dojo-course-3-the-full-orchard-core-tutorial

**Reply by hishamco** [reply]
What if you use ContentTypePicker to reference a content type into another

**Reply by ricocsharp** [reply]
Ok I will check for it thanks a lot !!

**Reply by hishamco** [reply]
Sorry I mean `ContentPickerField`
